File: 307-range-sum-query-mutable/307-range-sum-query-mutable.java
class NumArray {
    int[] arr;
    int sum = 0;
    public NumArray(int[] nums) {
        arr = nums;
        for(int i : nums){
            sum += i;
        }
    }
    
    public void update(int index, int val) {
        int diff = val - arr[index];
        sum += diff;
        arr[index] = val;
    }
    
    public int sumRange(int left, int right) {
        int l = 0;
        for(int i=0; i<left; i++) l += arr[i];
        for(int i = right + 1; i < arr.length; i++) l += arr[i];
        return sum - l;
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * obj.update(index,val);
 * int param_2 = obj.sumRange(left,right);
 */

File: 307-range-sum-query-mutable/307-range-sum-query-mutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.arr = nums
        self.s = sum(nums)

    def update(self, index: int, val: int) -> None:
        diff = val - self.arr[index]
        self.s += diff
        self.arr[index] = val

    def sumRange(self, left: int, right: int) -> int:
        l = sum([self.arr[i] for i in range(0, left)])
        r = sum([self.arr[i] for i in range(right+1, len(self.arr))])
        return self.s - l - r

# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# obj.update(index,val)
# param_2 = obj.sumRange(left,right)